packages/vite/src/client.ts

```diff
@@ -172,6 +172,20 @@ export async function buildClient (ctx: ViteBuildContext) {
     output: {
       chunkFileNames: ctx.nuxt.options.dev ? undefined : fileNames,
       entryFileNames: ctx.nuxt.options.dev ? 'entry.js' : fileNames,
+      sourcemapPathTransform (relativeSourcePath, sourcemapPath) {
+        // client build is running in a temporary build directory, like `.nuxt/dist/client`
+        // so we need to transform the sourcemap path to be relative to the final build directory
+        // which is normally `.output/public`
+        if (relativeSourcePath.startsWith('..')) {
+          const absoluteSourcePath = resolve(dirname(sourcemapPath), relativeSourcePath)
+          const newRelativePath = relative(
+            join(ctx.config.root!, '.output/public', ctx.nuxt.options.app.buildAssetsDir),
```

Comment:
we shouldn't hard-code `.output/public` as this will depend on the nitro preset used - we can get this from the resolved nitro options.